src/arithmetic.jl

```diff
@@ -15,18 +15,10 @@ for T in (:Taylor1, :TaylorN)
         =={T<:Number,S<:Number}(a::$T{T}, b::$T{S}) = ==(promote(a,b)...)
 
         function =={T<:Number}(a::$T{T}, b::$T{T})
-            la = a.order+1
-            lb = b.order+1
-            if a.order == b.order
-                return all( a.coeffs .== b.coeffs )
-            elseif a.order < b.order
-                res1 = all( a[1:la] .== b[1:la] )
-                res2 = iszero(b[la+1:lb])
-            else a.order > b.order
-                res1 = all( a[1:lb] .== b[1:lb] )
-                res2 = iszero(a[lb+1:la])
+            if a.order != b.order
```

Comment:
I suggest putting the `a.order != b.order` test inside `fixorder`, and just doing `a,b = fixorder(a, b)` here.